Dockerfile

```diff
@@ -31,47 +24,73 @@ ARG PG_MAJOR=13
+    cp "$source" /etc/apt/sources.list; \
+    rm -fr /tmp/sources
+
 # Make sure we're as up-to-date as possible, and install the highlest level dependencies
-RUN apt-get update && apt-get upgrade -y \
-    && apt-get install -y ca-certificates curl gnupg1 gpg gpg-agent locales lsb-release wget unzip
+RUN set -eux; \
+    apt-get update; \
+    apt-get upgrade -y; \
+    apt-get install -y ca-certificates curl gnupg1 gpg gpg-agent locales lsb-release wget unzip
 
 RUN mkdir -p /build/scripts
 RUN chmod 777 /build
 WORKDIR /build/
 
-RUN wget -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor --output /usr/share/keyrings/postgresql.keyring
+RUN curl -Ls https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor --output /usr/share/keyrings/postgresql.keyring
 RUN for t in deb deb-src; do \
         echo "$t [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/postgresql.keyring] http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -s -c)-pgdg main" >> /etc/apt/sources.list.d/pgdg.list; \
     done
 
-RUN apt-get clean
-RUN apt-get update
+# timescaledb-tune, as well as timescaledb-parallel-copy
+RUN curl -Ls https://packagecloud.io/timescale/timescaledb/gpgkey | gpg --dearmor --output /usr/share/keyrings/timescaledb.keyring
+RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/timescaledb.keyring] https://packagecloud.io/timescale/timescaledb/ubuntu/ $(lsb_release -cs) main" > /etc/apt/sources.list.d/timescaledb.list
 
 # The following tools are required for some of the processes we (TimescaleDB) regularly
 # run inside the containers that use this Docker Image
 # awscli is useful in many situations, for example, to list backup buckets etc
-RUN apt-get install -y less jq strace procps awscli
+RUN set -eux; \
+    apt-get update; \
+    apt-get upgrade -y; \
+    apt-get install -y \
+        less jq strace procps awscli vim-tiny gdb gdbserver dumb-init daemontools \
+        postgresql-common pgbouncer pgbackrest lz4 libpq-dev libpq5 pgtop libnss-wrapper gosu; \
+    curl -Lso /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_"$(dpkg --print-architecture)"; \
+    chmod 755 /usr/local/bin/yq
 
-# For debugging it is very useful if the Docker Image contains gdb(server). Even though it is
-# not expected to be running gdb in a live instance often, it simplifies getting backtraces from
-# containers using this image
-RUN apt-get install -y gdb gdbserver
+# forbid creation of a main cluster when package is installed
+RUN sed -ri 's/#(create_main_cluster) .*$/\1 = false/' /etc/postgresql-common/createcluster.conf
 
 # The next 2 instructions (ENV + RUN) are directly copied from https://github.com/rust-lang/docker-rust/blob/d534735bae832da4c60ddf799a8dfbefa9939020/1.67.0/bullseye/Dockerfile
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.65.0
+    RUST_VERSION=1.66.1
```

Comment:
Absolutely, and it does that mostly. Timescaledb, promscale, and toolkit builds happen in `build_scripts/install_extensions`. If it can find the artifact for rust things, it installs them, otherwise it'll build them. Timescaledb is always built though, since we use different compilation arguments than the .debs that are available.